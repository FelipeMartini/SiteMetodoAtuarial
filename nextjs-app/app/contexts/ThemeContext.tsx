/**
 * Contexto de Tema Unificado
 * Sistema moderno de gerenciamento de temas para a aplicação
 */
'use client';

import React, { createContext, useContext, useState, useEffect, ReactNode } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import { Theme, ThemeName, getThemeByName, getNextTheme, lightTheme } from '../../styles/themes';

interface ThemeContextProps {
  currentTheme: Theme;
  themeName: ThemeName;
  toggleTheme: () => void;
  setTheme: (name: ThemeName) => void;
  isDarkMode: boolean;
}

const ThemeContext = createContext<ThemeContextProps | undefined>(undefined);

interface ThemeProviderProps {
  children: ReactNode;
}

export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
  const [themeName, setThemeName] = useState<ThemeName>('light');
  const [currentTheme, setCurrentTheme] = useState<Theme>(lightTheme);

  // Carrega tema do localStorage na inicialização
  useEffect(() => {
    const savedTheme = localStorage.getItem('theme') as ThemeName;
    if (savedTheme && (savedTheme === 'light' || savedTheme === 'dark')) {
      setThemeName(savedTheme);
      setCurrentTheme(getThemeByName(savedTheme));
    }
  }, []);

  // Salva tema no localStorage quando muda
  useEffect(() => {
    localStorage.setItem('theme', themeName);
    setCurrentTheme(getThemeByName(themeName));
  }, [themeName]);

  const toggleTheme = () => {
    const nextTheme = getNextTheme(themeName);
    setThemeName(nextTheme.name as ThemeName);
  };

  const setTheme = (name: ThemeName) => {
    setThemeName(name);
  };

  const isDarkMode = themeName === 'dark';

  const value: ThemeContextProps = {
    currentTheme,
    themeName,
    toggleTheme,
    setTheme,
    isDarkMode,
  };

  return (
    <ThemeContext.Provider value={value}>
      <StyledThemeProvider theme={currentTheme}>
        {children}
      </StyledThemeProvider>
    </ThemeContext.Provider>
  );
};

export const useTheme = (): ThemeContextProps => {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error('useTheme deve ser usado dentro de um ThemeProvider');
  }
  return context;
};

// Alias para compatibilidade com código legado
export const useTema = useTheme;
export const useUtilsTema = useTheme;
