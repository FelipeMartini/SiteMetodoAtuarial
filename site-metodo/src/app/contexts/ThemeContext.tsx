// Arquivo movido para @core/theme/ContextoTema.tsx
// Remover este arquivo se não houver dependências legadas.
