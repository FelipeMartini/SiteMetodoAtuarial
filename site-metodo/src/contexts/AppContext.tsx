// Exemplo de AppContext
import { createContext } from "react";
export const AppContext = createContext({});
