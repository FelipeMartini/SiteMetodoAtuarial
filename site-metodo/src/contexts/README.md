# Contextos Globais

- Contextos de autenticação, tema, permissões, etc.
