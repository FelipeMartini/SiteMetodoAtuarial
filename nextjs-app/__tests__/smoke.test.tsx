// Teste de fumaça: verifica se o site carrega sem erros
import React from 'react';
import { render } from '@testing-library/react';
import Home from '../app/page';
import { ThemeProvider } from '../app/contexts/ThemeContext';

describe('Smoke Test', () => {
  it('deve renderizar a página inicial sem erros e exibir o título', () => {
    const { getByText, asFragment } = render(
      <ThemeProvider>
        <Home />
      </ThemeProvider>
    );
    // Verifica se o título principal aparece
    expect(getByText(/Método Atuarial/i)).toBeInTheDocument();
    // Snapshot para garantir que a estrutura principal não quebrou
    expect(asFragment()).toMatchSnapshot();
  });
});
