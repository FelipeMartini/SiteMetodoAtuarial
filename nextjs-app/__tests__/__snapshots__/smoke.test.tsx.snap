// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Smoke Test deve renderizar a p√°gina inicial sem erros e exibir o t√≠tulo 1`] = `
<DocumentFragment>
  <div
    class="sc-jJLAfE ifatkK"
  >
    <section
      class="sc-dYwGCk fjOdRu"
    >
      <div
        class="sc-hwkwBN nLwIx"
        style="text-align: center;"
      >
        <div>
          <div
            class="sc-kNOymR hvwBkq"
            style="font-size: 3rem; margin-bottom: 1rem;"
          >
            M√©todo Atuarial
          </div>
          <div
            class="sc-kNOymR eOJVkD"
            style="max-width: 600px;"
          >
            Consultoria especializada em solu√ß√µes atuariais, oferecendo excel√™ncia t√©cnica e inova√ß√£o para o mercado de previd√™ncia e seguros.
          </div>
        </div>
        <div
          class="sc-hwkwBN fgGIt"
        >
          <a
            href="/sobre"
            style="text-decoration: none;"
          >
            <button
              aria-label="Conhe√ßa Nossa Hist√≥ria"
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3 sc-eqYatC ZWTZT"
              data-slot="button"
              role="button"
              style="outline: none; box-shadow: 0 0 0 2px #2563eb;"
              tabindex="0"
              type="button"
            >
              Conhe√ßa Nossa Hist√≥ria
            </button>
          </a>
          <a
            href="/contato"
            style="text-decoration: none;"
          >
            <button
              aria-label="Solicitar Or√ßamento"
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3 sc-eqYatC iPSXie"
              data-slot="button"
              role="button"
              style="outline: none; box-shadow: 0 0 0 2px #2563eb;"
              tabindex="0"
              type="button"
            >
              Solicitar Or√ßamento
            </button>
          </a>
        </div>
      </div>
    </section>
    <section
      class="sc-dYwGCk fjOdRu"
    >
      <div
        class="sc-kNOymR kOnDQR"
        style="margin-bottom: 3rem;"
      >
        Nossos Servi√ßos
      </div>
      <div
        style="display: grid; grid-template-columns: repeat(auto-fit, minmax(300px, 1fr)); gap: 2rem; max-width: 1200px; margin: 0px auto;"
      >
        <div
          class="sc-kcLKEh PQfNo"
          style="cursor: default;"
        >
          <div
            style="display: flex; justify-content: center; align-items: center; margin-bottom: 8px;"
          >
            <span
              style="font-size: 3rem;"
            >
              üìä
            </span>
          </div>
          <h3
            class="sc-dntSTA gQABKC"
          >
            Consultoria Atuarial
          </h3>
          <p
            class="sc-kvnevz lkAtSo"
          >
            An√°lise de riscos e avalia√ß√£o de passivos previdenci√°rios
          </p>
        </div>
        <div
          class="sc-kcLKEh PQfNo"
          style="cursor: default;"
        >
          <div
            style="display: flex; justify-content: center; align-items: center; margin-bottom: 8px;"
          >
            <span
              style="font-size: 3rem;"
            >
              üìã
            </span>
          </div>
          <h3
            class="sc-dntSTA gQABKC"
          >
            Relat√≥rios Regulat√≥rios
          </h3>
          <p
            class="sc-kvnevz lkAtSo"
          >
            Atendimento √†s normas SUSEP, PREVIC e outros √≥rg√£os
          </p>
        </div>
        <div
          class="sc-kcLKEh PQfNo"
          style="cursor: default;"
        >
          <div
            style="display: flex; justify-content: center; align-items: center; margin-bottom: 8px;"
          >
            <span
              style="font-size: 3rem;"
            >
              üî¢
            </span>
          </div>
          <h3
            class="sc-dntSTA gQABKC"
          >
            Modelagem Matem√°tica
          </h3>
          <p
            class="sc-kvnevz lkAtSo"
          >
            Desenvolvimento de modelos atuariais customizados
          </p>
        </div>
      </div>
    </section>
    <section
      class="sc-dYwGCk fjOdRu"
    >
      <div
        class="sc-lgpSej cLwtrt"
        style="text-align: center; max-width: 800px; margin: 0px auto;"
      >
        <div
          class="sc-hwkwBN jksRbC"
        >
          <div
            class="sc-kNOymR dEIdFp"
          >
            Pronto para transformar seu neg√≥cio?
          </div>
          <div
            class="sc-kNOymR gUBsDM"
          >
            Entre em contato conosco e descubra como nossas solu√ß√µes atuariais podem otimizar seus resultados e garantir conformidade regulat√≥ria.
          </div>
          <a
            href="/contato"
            style="text-decoration: none;"
          >
            <button
              aria-label="Fale Conosco"
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3 sc-eqYatC ZWTZT"
              data-slot="button"
              role="button"
              style="outline: none; box-shadow: 0 0 0 2px #2563eb;"
              tabindex="0"
              type="button"
            >
              Fale Conosco
            </button>
          </a>
        </div>
      </div>
    </section>
    <section
      class="sc-dYwGCk hDlmKL"
    >
      <div
        class="sc-lgpSej cLwtrt"
        style="text-align: center;"
      >
        <div
          class="sc-kNOymR bULvpF"
        >
          Tema atual: Claro ‚òÄÔ∏è
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
